src/utils/websocket.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Utility module for websocket.
+ * @module utils/websocket
+ */
+import events from '../events';
+
+const messageIdsReceived = {};
+
+export function onMessageReceivedInternal(instance, msg) {
+    const messageId = msg.MessageId;
+    if (messageId) {
+        // message was already received via another protocol
+        if (messageIdsReceived[messageId]) {
+            return;
+        }
+
+        messageIdsReceived[messageId] = true;
+    }
+
+    if (msg.MessageType === "UserDeleted") {
+        instance._currentUser = null;
+    }
+    else if (msg.MessageType === "UserUpdated" || msg.MessageType === "UserConfigurationUpdated") {
```

Comment:
Since you're moving these might as well fix the annoying C# syntax in a few places.